### ISSUE TITLE
pilot distribute ignoring distribute_external flag

### ISSUE THREAD

**Issue by fer662** [fastlane/fastlane#5990]
### Issue Description

If you have an already processed build and you wanted to distribute it internally, you would:
`pilot distribute -a <bundle_id> --distribute_external false`

However, this doesn't work

> [09:03:27]: Distributing build 1.0(60) from Internal -> External
> [09:03:27]: Distributing new build to testers
> 
> Looking for related GitHub issues on fastlane/fastlane...
> 
> [09:03:32]: Error finding relevant GitHub issues: undefined method `map' for nil:NilClass
> 
> [!] The request could not be completed because:
>   Beta App Description is required. Feedback emails must be in the format: contact@example.com. Feedback emails are required. The marketing URL you entered is invalid. Make sure you entered it correctly and try again. Marketing URLs are required. The phone number is missing. The email must be in the format: contact@example.com. The email is required. Beta App Description is required. Feedback emails must be in the format: contact@example.com. Feedback emails are required. The marketing URL you entered is invalid. Make sure you entered it correctly and try again. Marketing URLs are required. The email must be in the format: contact@example.com. The email is required. The phone number is missing. Validation errors

All these errors are showing up because it's the info needed for an external distribution. Now if you check commands_generator.rb for the distribute command definition around line 70:

> ```
>   command :distribute do |c|
>     c.syntax = "pilot distribute"
>     c.description = "Distribute a previously uploaded binary to Apple TestFlight"
>     c.action do |args, options|
>       config = FastlaneCore::Configuration.create(Pilot::Options.available_options, convert_options(options))
>       config[:distribute_external] = true
>       Pilot::BuildManager.new.distribute(config)
>     end
>   end
> ```

You'll see that this flag is forced to be true. I tried removing this force and it's working fine for me. Is there any reason for the force being there? I'll make a pull request if not, since it's a trivial fix.
### Environment

fastlane 1.102.0
rvm 1.27.0

**Comment by TKBurner** [comment]
@fer662 Do you have at least one external tester setup for this app? That is a requirement on Apple's side. Thanks!

**Comment by fer662** [comment]
@TKBurner yes, sorry. The discussion was mostly going on on the pull request so I missed this.

**Comment by TKBurner** [comment]
@fer662 Do you have deliver setup for this project? If you don't can you run deliver init and let me know if you still see these metadata errors? Thanks!

**Comment by anantsaraswat** [comment]
@TKBurner I am very new to Fastlane, But I am using pilot to deliver the app, So i have used the following command, Which successfully uploaded my app on iTunes connect. 

pilot upload -u XXX@gmail.com -a XX.CCCC.DEV -i MYAPP.ipa --distribute_external false --wait_processing_interval 90 --verbose

Now I want it to enable automatically for internal testing. So I tried the following command, But it looks like it is trying to enable it for external users, instead of internal. 

pilot distribute --distribute_external false  -u XXX@gmail.com

Can you please let us know, What I am doing wrong, Or is there any better approach to achieve this, I just want my app to uploaded and automatically enable for internal testing.

**Comment by anantsaraswat** [comment]
@TKBurner  I got it running :dancer: , I think I have taken it so much seriously earlier. 

I have used the following command. 

pilot upload -u XXX@gmail.com -a XX.CCCC.DEV -p XXXXX67 -i MYAPP.ipa  --verbose 

Thanks @TKBurner

**Comment by TKBurner** [comment]
@anantsaraswat Awesome! So glad to hear you got up and running. Always feel free to reach out or search our issues  👍

**Comment by asalvo** [comment]
This tripped me up for awhile as well. Based on the documentation returned by "pilot --help" I would expect passing --distribute_external false to the distribute command to work. However, as pointed out distribute overrides the option. If people have to resort to looking at the source code, I think that's a sign that something could be improved :) 

If distribute will continue to ignore the configuration, then I would suggest the following:
- --help should be updated to reflect that the setting is ignored when passed to the distribute command. 
- There are a couple of places where the information message output to the screen should include the type of distribution (internal/external) as that would help make people more aware of what is going on.

So why do I want to call distribute after an upload? Well the first time I ran upload, I got an error back from the app store about the encryption setting not being defined.  However, my IPA was uploaded and available on ITunes connect for me to manually set to testing so I wanted to see if I could get it sent to testing from the command line. 

While I have fixed the encryption setting in my info.plist so my upload succeeds I'd still like to see pilot distribute support the distribute_externalFlag. It takes ~5 minutes for iTunes processing to finish (maybe it will take longer in the future) and I would prefer not to have my build agent tied up doing nothing. I had been planning on scheduling a follow-up build to check the processing status and do the distribution. 

For now I'll just have to accept that my build agent will be tied up for an extra ~5 minutes waiting for iTunes to finish processing. Maybe it will provide me with some extra justification to get more/faster build agents :)
